File: benchmarks/oopsla25/subjects/401.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f401(const char *s) {
    int i = 0;
    int ca = 0;
    int cb = 0;
    while (i < strlen(s)) {
        if (s[i] == 'a') {
            ca += 1;
        }
        if (s[i] == 'b') {
            cb += 1;
        }
        i++;
    }
    assert(ca == strlen(s) || cb == strlen(s));
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f401(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/401.py
#!/usr/bin/env python3

import sys

def f401(s: str):
  i = 0
  ca = 0
  cb = 0
  while i < len(s):
    if s[i] == "a":
      ca += 1
    if s[i] == "b":
      cb += 1
    i += 1
  assert ca == len(s) or cb == len(s)

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f401(my_string)
